src/main/java/seedu/address/model/person/ModuleRoleMap.java

```diff
@@ -0,0 +1,232 @@
+     *
+     * @param moduleCodes Array of module codes.
+     * @return whether the array contains all valid ModuleCode
+     */
+    public static boolean isValidModuleCodeArray(ModuleCode[] moduleCodes) {
+        requireNonNull(moduleCodes);
+        for (ModuleCode moduleCode: moduleCodes) {
+            if (moduleCode == null) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    /**
+     * Returns true if the given RoleType array does not contain null object.
+     *
+     * @param roleTypes Array of RoleType objects.
+     * @return whether the array contains all valid RoleType objects.
+     */
+    public static boolean isValidRoleTypeArray(RoleType[] roleTypes) {
+        requireNonNull(roleTypes);
+        for (RoleType roleType: roleTypes) {
+            if (roleType == null) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    /**
+     * Converts a given role type keyword to its corresponding RoleType Enum value.
+     *
+     * @param keyword Keyword representing a role type.
+     * @return corresponding RoleType Enum value.
+     */
+    public static RoleType roleTypeStringToEnum(String keyword) {
+        requireNonNull(keyword);
+        switch (keyword.toLowerCase()) {
+        case "": // Fall through
+        case "student":
+            return RoleType.STUDENT;
+        case "tutor": // Fall through
+        case "ta":
+            return RoleType.TUTOR;
+        case "professor": // Fall through
+        case "prof":
+            return RoleType.PROFESSOR;
+        default:
+            throw new IllegalArgumentException("Invalid role type keyword for conversion to Enum value");
+        }
+    }
+
+    /**
+     * Gets a filtered stream of ModuleCode objects based on the given RoleType Enum Value.
+     * Provides an easier way to query, search and manipulate modules.
+     *
+     * @param roleType RoleType enum value used to filter the ModuleCode
+     * @return A Stream of ModuleCode that are linked to a specific RoleType
+     */
+    public Stream<ModuleCode> getFilteredModuleCodes(RoleType roleType) {
+        requireNonNull(roleType);
+        return this.roles.entrySet().stream()
+                .filter(entry -> entry.getValue().equals(roleType))
+                .map(Map.Entry::getKey);
+    }
+
+    /**
+     * Gets the comma separated String representation of a list of moduleCodes based on
+     * the filtered Stream of ModuleCode.
+     *
+     * @param moduleCodeStream filtered Stream of module codes.
+     * @return comma separated String representation of a list of ModuleCode
+     */
+    public static String getModuleCodeString(Stream<ModuleCode> moduleCodeStream) {
+        requireNonNull(moduleCodeStream);
+        String[] moduleCodeStringArray = moduleCodeStream
+                .map(ModuleCode::toString)
+                .toArray(String[]::new);
+        return String.join(", ", moduleCodeStringArray);
```

Comment:
Can be simplified as such:
```
return moduleCodeStream
            .map(ModuleCode::toString)
            .collect(Collectors.joining(", "));
```

Reply:
Thanks for the suggestion! Will implement the change! Thank you!